src/AWS/Lambda/Context.hs

```diff
@@ -83,6 +83,11 @@ class HasLambdaContext r where
 instance HasLambdaContext LambdaContext where
   withContext = const
 
+-- TODO: This sticks around for backwards compatibility, and as a conevient-ish
+-- way to runReaderTLambdaContext.  In the long term though, all runtimes based
+-- on the ReaderT LambdaContext approach will be removed, and this instance
```

Comment:
Even `readerTRuntime :: (FromJSON event, ToJSON result) => (event -> ReaderT LambdaContext IO result) -> IO ()`? It seems safe.

Reply:
Ah, right, good callout, since we just can't have the Reader constraint on the _resulting_ Monad, but the runtime itself can't push it into a Reader for you at the top of the stack.  I'll try to phrase this better...

Reply:
Which I think forces it to be the outermost constraint unless you want to start asking the caller to provide a natural transformation between the inner and outer monads: `(event -> ReaderT LambdaContext m result) -> m ()` (ignoring constraints). But this is morally the same as taking `LambdaContext` as a param, except it makes `const`/`withoutContext` harder to use as a way to ignore the context when you don't care about it.

I think it can go away, but maybe it's better to keep it around for backwards compat. Instead, maybe write something like: "In the longer term, all runtimes that use a (MonadReader r m, HasLambdaContext r) constraint will be removed, and..."

Reply:
Agreed, there's no reason really to drop `readerTRuntime`.  Check out the recent push and let me know if that describes it better!

Reply:
Yes, that's better than what I wrote.

Reply:
Whatever gets the job done!  And it's certainly better than my first pass after review 😆